### PR TITLE
feat: remove the implementation of dev runserver

### PR DESCRIPTION
**Pull request by Carlos-Muniz**
This PR removes the implementation of `tutor dev runserver`. It is being removed in favor of using `tutor dev start`.

This PR should be merged last after each of the following have been merged:

- https://github.com/overhangio/tutor-ecommerce/pull/31
- https://github.com/overhangio/tutor-forum/pull/8
- https://github.com/overhangio/tutor-notes/pull/13
- https://github.com/overhangio/tutor-xqueue/pull/9

Closes https://github.com/overhangio/2u-tutor-adoption/issues/26

### CODE REVIEW

**Comment by Carlos-Muniz** [comment]
@kdmccormick @regisb This is ready for review.

**Review by kdmccormick — CHANGES_REQUESTED** [review]
Code changes look good, though you've got a linting error.

Make sure you add a CHANGELOG entry as described in the [contribution docs](https://docs.tutor.overhang.io/tutor.html?highlight=contrib#contributing-to-tutor).

**Comment by Carlos-Muniz** [comment]
@kdmccormick I've addressed the linting error and added an entry to `CHANGELOG.md`

**Comment by kdmccormick** [comment]
> Where “TYPE” is either “Bugfix”, “Improvement”, “Feature” or “Security”. You should add an explosion emoji (”💥”) before “[TYPE]” if you are making a breaking change.

This is a breaking change, and I think "Improvement" is the most appropriate type.

**Comment by Carlos-Muniz** [comment]
@kdmccormick updated with your suggestion 👍

**Review by kdmccormick — APPROVED** [review]
LGTM

**Review by regisb — COMMENTED** [review]
I'd rather not make such breaking changes between two releases. I mean: sometimes we do make breaking changes between releases, but this one seems slightly too important for my taste. Plus, there is no rush to remove the `runserver` command, right? I think that the obsolete warning should be sufficient for now.

Do you think this PR could wait until Olive? Or maybe you can make this change on top of the nightly branch?

**Comment by kdmccormick** [comment]
@regisb Carlos is just doing what we asked him to do :) From https://github.com/overhangio/2u-tutor-adoption/issues/26 :
> In order to give folks time to stop using runserver, wait until Tutor v14 is released.

But certainly, there's no rush on this. We could definitely wait until Olive or merge this to Nightly, whichever you prefer.

**Comment by kdmccormick** [comment]
@regisb I've re-opened this PR on top of nightly: https://github.com/overhangio/tutor/pull/718

**Pull request closed without merging**
